File: safelima_app/app/models/app_feedback.py
from sqlalchemy import Column, Integer, String, Text, Numeric, DateTime, ForeignKey, UniqueConstraint
from sqlalchemy.orm import relationship
from datetime import datetime
from app.database.database import Base

class AppFeedback(Base):
    __tablename__ = "app_feedback"
    
    id = Column(Integer, primary_key=True, index=True)
    citizen_id = Column(Integer, ForeignKey("citizens.id", ondelete="CASCADE"), nullable=False, index=True, unique=True)
    estrellas = Column(Integer, nullable=False)
    comentario = Column(Text, nullable=True)
    fecha = Column(DateTime, default=datetime.utcnow)

    # Relación
    citizen = relationship("Citizen")

File: safelima/lib/models/app_feedback.dart
class AppFeedback {
  final int? id;
  final int citizenId;
  final int estrellas;
  final String? comentario;
  final DateTime? fecha;

  AppFeedback({
    this.id,
    required this.citizenId,
    required this.estrellas,
    this.comentario,
    this.fecha,
  });

  factory AppFeedback.fromJson(Map<String, dynamic> json) {
    return AppFeedback(
      id: json['id'],
      citizenId: json['citizen_id'],
      estrellas: json['estrellas'],
      comentario: json['comentario'],
      fecha: json['fecha'] != null ? DateTime.parse(json['fecha']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'citizen_id': citizenId,
      'estrellas': estrellas,
      'comentario': comentario,
    };
  }
}
